### ISSUE TITLE
 Microsoft 365 learning pathways has a configuration issue. Intermittent problem.

### ISSUE THREAD

**Issue by LarsKragKongsgaard** [pnp/custom-learning-office-365#747]
>**PROVISIONING NOTE**: If you are having issues with the SharePoint look book while trying to provision M365 learning pathways please do not post that issue to this repo as we cannot help you here. Please post all questions and issues for the SharePoint look book to [SharePoint/sp-provisioning-service GitHub Issues List](https://github.com/SharePoint/sp-provisioning-service/issues).

>**WEB PART NOT LOADING**: If you are getting the error: "Microsoft 365 learning pathways has a configuration issue. Ask your administrator for assistance". Please review the [Diagnosing Pre-Check](https://github.com/pnp/custom-learning-office-365/blob/main/installation/DiagnosisPreCheck.md) guidance before submitting your bug.

## Describe the bug

The Microsoft 365 learning pathways search is not loading in the the learning hub: https://mytenant.sharepoint.com/sites/M365LP/

### To Reproduce

Go to https://mytenant.sharepoint.com/sites/M365LP/SitePages/Microsoft-365-products.aspx

### Expected behavior

The Microsoft 365 learning pathways search should appear and allow me to search products

### Screenshots

Please see attached

### Learning Pathways version number

Learning Pathways App version 4.3.1.0

### Additional context

The problem is intermittent - comes and goes in a work day - really disturbing
[1-oneiig.sharepoint.com-1688975741010.log](https://github.com/pnp/custom-learning-office-365/files/12002376/1-oneiig.sharepoint.com-1688975741010.log)
[oneiig.sharepoint.com-1688975689791.log](https://github.com/pnp/custom-learning-office-365/files/12002377/oneiig.sharepoint.com-1688975689791.log)
[working - oneiig.sharepoint.com.zip](https://github.com/pnp/custom-learning-office-365/files/12002378/working.-.oneiig.sharepoint.com.zip)
[Not working - oneiig.sharepoint.com.zip](https://github.com/pnp/custom-learning-office-365/files/12002382/Not.working.-.oneiig.sharepoint.com.zip)

**Comment by dcashpeterson** [comment]
@LarsKragKongsgaard let me take a look at this. I can see from the logs that it is blocking the manifest due to a CORS errors. Are there any steps we can take to reproduce this regularly? You said it is intermittent, is it consistent with specific users or is it all users?

**Comment by LarsKragKongsgaard** [comment]
Hi Derek,

Thank you for your response. It seems to be intermittent for all users.
We have been testing for a while, me and Malin Pettersson ***@***.******@***.***>

It is easy to reproduce because it seems to happen daily.
Sometimes, when I have the problem, the I disconnect my VPN connection and then it seems to work.
It is anyway same problem from remote work or from office, it seems.

Kr / Lars Krag

**Comment by dcashpeterson** [comment]
@LarsKragKongsgaard I am assigning this to @v-prigunasek to continue the investigation but can you please confirm the following?  

1. You have a custom page called Microsoft-365-products
2. You have an explanation of how to do searching
3. There is a learning pathways web part below
4. The web part loads fine until the user tries to search or the web part doesn't load at all?

**Comment by LarsKragKongsgaard** [comment]
Hi Derek,

Thank you for your attention. Please see inline comments below.

Kr / Lars Krag

**Comment by v-prigunasek** [comment]
Hi @LarsKragKongsgaard,

Can you please respond to the issue here in this GitHub repo rather than email? I haven't received the screenshots or links that you have shared in the email response. So please share those details here. Thanks!

**Comment by LarsKragKongsgaard** [comment]
Sorry, yes, I will do it. Kr / Lars Krag

**Comment by LarsKragKongsgaard** [comment]
1.  You have a custom page called Microsoft-365-products
[Lars Krag] Yes, it is a SharePoint Communication site with the Leaning pathways: Customize learning pathways | Microsoft Learn<https://learn.microsoft.com/en-us/office365/customlearning/custom_overview>

  2.  You have an explanation of how to do searching
[Lars Krag] Yes, there is a link to the M365 Learning Pathways App search from the frontpage and with a brief explanation.
[Customize learning pathways | Microsoft Learn](https://learn.microsoft.com/en-us/office365/customlearning/custom_overview)

Today, it seems to be working partly at least (search function, but not update playlist function).
See attached: Picture1.jpg
![Picture1](https://github.com/pnp/custom-learning-office-365/assets/75684369/20fab433-bb1f-4c17-acbe-5ea570c8dcfc)

Update from "gear wheel" not working:
See attached Picture2.jpg
![Picture2](https://github.com/pnp/custom-learning-office-365/assets/75684369/e2146fcf-1b76-4c6d-9eb9-d59d8636c77e)

  3.  There is a learning pathways web part below
[Lars Krag] Yes

  4.  The web part loads fine until the user tries to search or the web part doesn't load at all?
[Lars Krag] The web part does not load at all.
See also screenshot of configuration: Picture3.jpg
![Picture3](https://github.com/pnp/custom-learning-office-365/assets/75684369/d4bf53e2-8ad5-475f-b2b2-1fdc2b7e4f59)

You attention is highly appreciated.

Kr / Lars Krag

**Comment by LarsKragKongsgaard** [comment]
I updated. Now I learned a bit about these comments 😊 Kr / Lars Krag

From: Lars Krag Kongsgaard
Sent: Wednesday, July 12, 2023 2:58 PM
To: pnp/custom-learning-office-365 ***@***.***>
Subject: RE: [pnp/custom-learning-office-365] Microsoft 365 learning pathways has a configuration issue. Intermittent problem. (Issue #747)

Sorry, yes, I will do it. Kr / Lars Krag

**Comment by v-prigunasek** [comment]
Thanks for sharing the details. Let me analyze it and get back to you with my findings. Thanks!

**Comment by v-prigunasek** [comment]
Hi @LarsKragKongsgaard ,

From the logs, I see the Manifest is getting blocked due to a CORS restriction. Also, I see that you have mentioned, it happens when you are using the VPN. Can you please check with your network team and make sure the below domain or URL is white-listed and not blocked by any means in your network?
https://pnp.github.io/custom-learning-office-365/

Unfortunately, we can't really troubleshoot CORS related issues from our end.

Request you to refer similar issues reported by users related to CORS policy.
https://github.com/pnp/custom-learning-office-365/issues/403
https://github.com/pnp/custom-learning-office-365/issues/656

In case more details are required please let us know.

**Comment by LarsKragKongsgaard** [comment]
Let me get back to you. Kr /  Lars Krag

**Comment by LarsKragKongsgaard** [comment]
Thanks Derik [@dcashpeterson] I will check this also with the network team. Let me get back to you on any conditional access policy blocking the SharePoint Online app catalog in our tenant. Kr / Lars Krag

**Comment by LarsKragKongsgaard** [comment]
BTW Derek [@dcashpeterson] the problem still persists. Please see attached.
![Screenshot 2023-08-02 165539](https://github.com/pnp/custom-learning-office-365/assets/75684369/3a5b12b7-4a02-40b5-9b83-cec58c64fc87)

**Comment by v-prigunasek** [comment]
Hi @LarsKragKongsgaard , 

Were you able to check with your internal network team to check whether this URL - https://pnp.github.io/custom-learning-office-365/ is blocked by any means? 

Can you also please check if you are able to access this URL directly in the browser - [pnp.github.io/custom-learning-office-365/learningpathways/v4/manifest.json](https://pnp.github.io/custom-learning-office-365/learningpathways/v4/manifest.json)

**Comment by LarsKragKongsgaard** [comment]
Hi @v-prigunasek, I am still working on this one and it is taking far too long, sorry to say. I aim to get back to you at the end of this week. I am able to access the manifest file from your message directly. The strange thing is that the problems seems to be intermittent, sometimes the problem is there and sometimes not. Kr / Lars Krag

**Comment by LarsKragKongsgaard** [comment]
Still checking with the network team. Here is a sbip from the browser logfiles: 
"
CustomLearningAdmin.aspx:1  Access to fetch at 'https://pnp.github.io/custom-learning-office-365/learningpathways/v4/manifest.json' from origin 'https://oneiig.sharepoint.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
pnp.github.io/custom-learning-office-365/learningpathways/v4/manifest.json:1 
        
        
        Failed to load resource: net::ERR_FAILED
custom-learning-admin-web-part_ddcc24dbbbeac735e927.js:1  Message: 🎓 M365LP:InitService (getManifest) - TypeError: Failed to fetch
"

**Comment by v-asshrirao** [comment]
@LarsKragKongsgaard , Thank you for sharing the logs.
Unfortunately, we can't really troubleshoot CORS related issues from our end.

In case mode details are required please let us know.

**Comment by dcashpeterson** [comment]
@LarsKragKongsgaard I'm closing this as it seems like a CORS issues related to your network. As @v-asshrirao mentioned if you can get the more info from your networking team please open a new issue and reference this one.
